### PR TITLE
(DOCSP-27598): Clean up web auth examples

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

Clean up auth examples

### Jira

- https://jira.mongodb.org/browse/DOCSP-27598

### Staged Changes

- [Authentication - Web SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-27598/web/authenticate)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/examples/generated/web/authentication-log-in-and-out.test.snippet.builtin-apple-oauth.js [hunk 1]
```diff
@@ -4,11 +4,11 @@ const redirectUri = "https://app.example.com/handleOAuthLogin";
 const credentials = Realm.Credentials.apple(redirectUri);
 
 // Calling logIn() opens an Apple authentication screen in a new window.
-app.logIn(credentials).then((user) => {
-  // The logIn() promise will not resolve until you call `handleAuthRedirect()`
-  // from the new window after the user has successfully authenticated.
-  console.log(`Logged in with id: ${user.id}`);
-});
+const user = app.logIn(credentials);
```
Comment (nlarew) [inline source/examples/generated/web/authentication-log-in-and-out.test.snippet.builtin-apple-oauth.js:7]:
```suggestion
const user = await app.logIn(credentials);
```

Do we need to await the Promise?

File: source/examples/generated/web/authentication-log-in-and-out.test.snippet.builtin-apple-oauth.js [hunk 2]
```diff
@@ -4,11 +4,11 @@ const redirectUri = "https://app.example.com/handleOAuthLogin";
...
+
+// The logIn() promise will not resolve until you call `handleAuthRedirect()`
+// from the new window after the user has successfully authenticated.
+console.log(`Logged in with id: ${user.id}`);
```
Comment (nlarew) [inline source/examples/generated/web/authentication-log-in-and-out.test.snippet.builtin-apple-oauth.js:11]: Can we split this into two code snippets? It's not totally clear to the skimmer that ``handleAuthRedirect()`` happens in another window / URL. Similar comment for the other OAuth providers.

Also, we have the same codeblock for Facebook but an entirely different format for Google. Should we unify these?

File: source/web/authenticate.txt [hunk 3]
```diff
@@ -195,7 +190,7 @@ is a Google SDK that lets users sign up to or log in to a website with one click
     This example shows how to set up Google Authentication with App Services in
     a very basic web app. The app only contains an ``index.html`` file.
```
Comment (nlarew) [inline source/web/authenticate.txt:192]: In the sentence above, it's a bit strange that we say "The app only contains an ``index.html`` file." and then show a tree that also includes `auth.html`.

File: source/web/authenticate.txt [hunk 4]
```diff
@@ -272,7 +267,7 @@ require you to install a Google SDK. The built-in flow follows three main steps:
       ├── auth.html
       ├── index.html
```
Comment (nlarew) [inline source/web/authenticate.txt:268]: For the last entry of the tree output can we use `└` instead of `├`
```suggestion
      └── index.html
```

**Review by nlarew — COMMENTED** [review]
Code changes look good, a couple suggestions

File: source/web/authenticate.txt (hunk 3 above)
Comment (mongodben) [reply source/web/authenticate.txt:192]: that's a separate code example (you need to expand the code on github to see).

File: source/examples/generated/web/authentication-log-in-and-out.test.snippet.builtin-apple-oauth.js (hunk 2 above)
Comment (mongodben) [reply source/examples/generated/web/authentication-log-in-and-out.test.snippet.builtin-apple-oauth.js:11]: i thought about doing that in the first pass but didn't b/c of fact it was pretty outside scope of original ticket + some laziness. but we already got outside the scope of the ticket so might as well go for it.

Comment (mongodben) [reply source/examples/generated/web/authentication-log-in-and-out.test.snippet.builtin-apple-oauth.js:11]: and in the spirit of scope creep, i think i'll just throw in https://jira.mongodb.org/browse/DOCSP-28363 w this ticket as well.

File: source/web/authenticate.txt (hunk 3 above)
Comment (nlarew) [reply source/web/authenticate.txt:192]: Oh duh, my bad

**Review by nlarew — APPROVED** [review]

File: source/examples/generated/web/authentication-log-in-and-out.test.snippet.builtin-apple-oauth.js (hunk 2 above)
Comment (mongodben) [reply source/examples/generated/web/authentication-log-in-and-out.test.snippet.builtin-apple-oauth.js:11]: >90% sure https://jira.mongodb.org/browse/DOCSP-28363 doesn't require any work. waiting for final confirmation from the eng who opened that ticket. but merging this now.

**Pull request merged**